ext/gd/gd.c

```diff
@@ -4046,13 +4044,13 @@ PHP_FUNCTION(imageaffine)
 
 
 	src = php_gd_libgdimageptr_from_zval_p(IM);
-
-	if ((nelems = zend_hash_num_elements(Z_ARRVAL_P(z_affine))) != 6) {
+	uint32_t nelems = zend_hash_num_elements(Z_ARRVAL_P(z_affine));
+	if (nelems != 6) {
 		zend_argument_value_error(2, "must have 6 elements");
 		RETURN_THROWS();
 	}
 
-	for (i = 0; i < nelems; i++) {
+	for (uint32_t i = 0; i < nelems; i++) {
```

Comment:
looking good so far, I think you can do the same stripping down you did above, i.e. we want 6 elements from the matrix.

Reply:
The main reason I use uint32_t here is that we are comparing it to `nelems` which is a `uint32_t`

Reply:
I meant more something like

```c
for (uint32_t i = 0; i < 6; i ++) ..
```

to save the variable like you did above.

Reply:
But I did this, (in a follow-up commit from the one you reviewed?)